crates/bevy_core_pipeline/src/contrast_adaptive_sharpening/node.rs

```diff
@@ -0,0 +1,129 @@
+use std::sync::Mutex;
+
+use crate::contrast_adaptive_sharpening::ViewContrastAdaptiveSharpeningPipeline;
+use bevy_ecs::prelude::*;
+use bevy_ecs::query::QueryState;
+use bevy_render::{
+    extract_component::ComponentUniforms,
+    render_graph::{Node, NodeRunError, RenderGraphContext, SlotInfo, SlotType},
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindingResource, Operations, PipelineCache,
+        RenderPassColorAttachment, RenderPassDescriptor, TextureViewId,
+    },
+    renderer::RenderContext,
+    view::{ExtractedView, ViewTarget},
+};
+
+use super::{CASUniform, ContrastAdaptiveSharpeningPipeline};
+
+pub struct ContrastAdaptiveSharpeningNode {
+    query: QueryState<
+        (
+            &'static ViewTarget,
+            &'static ViewContrastAdaptiveSharpeningPipeline,
+            // &'static CASUniform,
```

Comment:
I ended up shoving the uniforms into a Resource on extract (which, thinking about it, is wrong if we're going to let users have different settings per view) because I couldn't get it to accept the CASUniform struct when calling `create_bind_group` in `ContrastAdaptiveSharpeningNode::run()`.

Reply:
Check out `UniformComponentPlugin`. The revamp bloom PR has an example of using it, and it's probably used elsewhere in bevy. Basically you can use that plugin, and then in the node you add a component to the view query, which gives you a dynamic offset that you set on the render pass. Feel free to ask me about it in more detail on discord.

Reply:
I was already using `UniformComponentPlugin`, I just wasn't dynamically indexing into the resource containing it.

Fixing this also fixed a potential future bug where using it on multiple cameras with different values was broken.